pkg/controller/watch/event_handler.go

```diff
@@ -0,0 +1,94 @@
+package watch
+
+import (
+	"go.uber.org/zap"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/kube"
+)
+
+type ResourceEventHandler struct {
```

Comment:
as discussed p2p - it should be ok to use the `AtlasResourceEventHandler` here and "override" the delete handle